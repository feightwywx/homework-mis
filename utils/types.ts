import { classRow, studentRow } from "./user";

export type User = {
  actual_id: string;   // 登录名
  isLoggedIn: boolean;
  name: string;   // 登录名，如果有显示名称则会被覆盖
  userType: UserType;
  token: string;
};

export type UserType = "student" | "teacher";

export interface StudentHomework extends Homework {
  completed: boolean;
}

export interface Homework {
  id: number;
  title: string;
  assignment: string;
  time: string;
  deadline: string;
  teacher: string;
  courseName: string;
  courseID: number;
}

export interface HomeworkDetail {
  title: string;
  assignment: string;
  time: string;
  deadline: string;
}

export interface HomeworkDetailContent {
  completed: boolean;
  accomplishment?: {
    time: string;
    content: string;
  };
  judge?: {
    score: number;
    comment: string;
  };
}

export interface HomeworkStudentDetail {
  detail: HomeworkDetail;
  content: HomeworkDetailContent;
}

export interface HomeworkTeacherDetailContent extends HomeworkDetailContent {
  contentId: number;
  studentName: string;
}

export interface HomeworkTeacherDetail {
  detail: HomeworkDetail;
  content: Array<HomeworkTeacherDetailContent>;
}

export type ResponseResult =
  | string
  | Homework[]
  | UpdateResult
  | studentRow[]
  | classRow[]
  | User;

export interface JsonResponse<T = ResponseResult> {
  code: number;
  result: T;
}

export interface UpdateResult {
  affected: number;
}

export interface Course {
  id: number;
  name: string;
  time: string;
  ended: boolean;
  teacherID: number;
  teacherName: string;
}

export interface UserDetail {
  actualID: string;
  name: string;
}

export type TeacherUserDetail = UserDetail;

export interface StudentUserDetail extends UserDetail {
  class: string;
}

export interface Exam {
  id: number;
  name: string;
  time: string;
  endtime: string;
  location: string;
  courseID: number;
  courseName: string;
}

export interface ExamResult {
  id: number;
  studentID: number;
  studentName: string;
  score: number | null;
}

export type ExamAssginRequestBody = {
  title: string;
  location: string;
  time: string;
  endtime: string;
  courseID: number;
}

export type HomeworkContentScore = {
  id: number;
  completed?: boolean;
  score?: number;
  homeworkID: number;
  homeworkTitle: string;
}

export type ExamContentScore = {
  id: number;
  score?: number;
  examID: number;
  examName: string;
}

export type HomeworkTotalScore = {
  studentID: number;
  studentName: string;
  completionRate: number;
  averageScore: number;
};

export type ExamTotalScore = {
  studentID: number;
  studentName: string;
  averageScore: number;
};
