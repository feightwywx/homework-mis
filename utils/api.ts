import { NextApiRequest } from "next";
import { JsonResponse, ResponseResult, UserType } from "./types";
import { getId } from "./user";

export function successResponse<T = ResponseResult>(result?: T) {
  return {
    code: 0, 
    result: result ?? ''
  } as JsonResponse<T>
}

export function failResponse(code: number, result?: ResponseResult) {
  return {
    code, 
    result: result ?? ''
  } as JsonResponse
}

export async function parseIdFromReqest(req: NextApiRequest, userType: UserType) {
  let token = undefined;
  if (req.session.user?.token) {
    token = req.session.user?.token
  } else {
    token = await req.body.token;
  }

  return await getId(userType, token);
}

export const statusCode = {
  ROUTER_PARAM_REQUIRED: 100,
  ROUTER_PARAM_INVALID: 1001,
  BODY_PARAM_REQUIRED: 101,
  URL_PARAM_REQUIRED: 102,
  TOKEN_INVALID: 200,
  NUL_QUERY_DATA: 300,
  NUL_QUERY_PARAM: 301,
  BAD_UPDATE: 400
}
