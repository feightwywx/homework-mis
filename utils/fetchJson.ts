export default async function fetchJson<JSON = unknown>(
  input: RequestInfo, init?: RequestInit
): Promise<JSON> {
  const res = await fetch(input, init);
  const data = await res.json();

  if (res.ok) return data;

  throw new Error(res.statusText);
}