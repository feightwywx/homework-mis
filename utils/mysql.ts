import { createPool } from 'mysql2/promise';
import { createConnection } from 'mysql2/promise';

export async function createMisConn() {
  return createConnection({
    host: 'localhost',
    user: 'root',
    password: '',
    database: 'homework_mis'
  })
}

export const sqlPool = createPool({
  host: 'localhost',
  user: 'root',
  password: '',
  database: 'homework_mis'
})

export async function getNowMySqlDateTime() {
  const date = new Date()
  return `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()} ${date.getHours()}:${date.getMinutes()}:${date.getSeconds()}`
}
