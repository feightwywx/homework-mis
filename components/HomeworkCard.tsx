import { Avatar, Card, Divider, Tag, Typography } from 'antd';
import { CheckCircleOutlined } from '@ant-design/icons';
import { UserOutlined } from '@ant-design/icons';
import parseMysqlDateTime from '../utils/parseTime';
import { Homework, StudentHomework } from '../utils/types';
import Link from 'next/link';
import useUser from '../utils/hooks/useUser';

const { Text } = Typography

export function HomeworkCard({ homework }: { homework: Homework | StudentHomework; }): JSX.Element {
  const time = parseMysqlDateTime(homework.time)
  const deadline = parseMysqlDateTime(homework.deadline);
  const currtime = new Date(Date.now())

  const { user } = useUser()

  return (
    <Card title={
      <>
        {homework.title}
        {
          ((homework as StudentHomework).completed
          && <Tag
            icon={<CheckCircleOutlined />}
            color='success'
            style={{
              marginLeft: 8,
              verticalAlign: 'middle'
            }}
          >已完成</Tag>) || (
            currtime > deadline
            && <Tag
            style={{
              marginLeft: 8,
              verticalAlign: 'middle'
            }}
          >已过期</Tag>
          )
        }
      </>}
      extra={<Link href={`homeworkDetail/${homework.id}`}>查看</Link>}
      style={{ marginBottom: 16, display: 'block' }}>
      <Text>发布时间：{time.toLocaleString()}</Text><br />
      <Text>截止时间：{deadline.toLocaleString()}</Text>
      <Divider style={{ margin: '16px 0px' }} />
      <div>
        <Avatar size={'small'} icon={<UserOutlined />} style={{ marginTop: -4 }} />
        <Text style={{ marginLeft: 12 }}><b>{homework.teacher}</b>布置的作业</Text>
      </div>
    </Card>
  );
}
