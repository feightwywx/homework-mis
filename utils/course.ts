import { createMisConnection } from "./mysql";
import { Course } from "./types";

type CourseRow = {
  id: number;
  name: string;
  time: string;
  ended: number;
  teacherID: number;
  teacherName: string;
}

export async function getStudentCourses(id: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT course.* , teacher.name AS teacherName
    FROM course
    JOIN course_student ON course.id = course_student.studentID 
    JOIN teacher ON course.teacherID = teacher.id
    WHERE course_student.studentID = ?;`,
    [id]
  );
  await conn.end();

  const courses = (rows as Array<CourseRow>).map((row) => {
    return {
      ...row,
      ended: row.ended === 0 ? false : true,
    } as Course;
  });

  return courses;
}

export async function getTeacherCourses(id: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT course.* , teacher.name AS teacherName
    FROM course
    JOIN teacher ON course.teacherID = teacher.id
    WHERE teacher.id = ?;`,
    [id]
  );
  await conn.end();

  const courses = (rows as Array<CourseRow>).map((row) => {
    return {
      ...row,
      ended: row.ended === 0 ? false : true,
    } as Course;
  });

  return courses;
}

export async function getCourseByID(id: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT course.* , teacher.name AS teacherName
    FROM course
    JOIN teacher ON course.teacherID = teacher.id
    WHERE course.id = ?;`,
    [id]
  );
  await conn.end();

  const courses = (rows as Array<CourseRow>).map((row) => {
    return {
      ...row,
      ended: row.ended === 0 ? false : true,
    } as Course;
  });

  return courses[0];
}

export async function getCourses() {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT course.*
    FROM course`
  );
  await conn.end();

  const courses = (rows as Array<CourseRow>).map((row) => {
    return {
      ...row,
      ended: row.ended === 0 ? false : true,
    } as Course;
  });

  return courses;
}
