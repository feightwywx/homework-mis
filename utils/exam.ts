import { ResultSetHeader } from "mysql2";
import { createMisConnection } from "./mysql";
import { Exam, ExamResult } from "./types";

type ExamRow = {
  id: number;
  name: string;
  time: string;
  endtime: string;
  location: string;
  courseID: number;
};

type ExamResultRow = {
  score: number | null;
};
type DetailedExamResultRow = ExamResultRow & {
  id: number;
  studentID: number;
  studentName: string;
};

export async function getExamsByCourseID(coid: number): Promise<Exam[]> {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT *
    FROM exam
    WHERE courseID=?`,
    [coid]
  );
  await conn.end();

  const exams = rows as Array<ExamRow>;

  return exams;
}

export async function getExamByID(eid: number): Promise<Exam> {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT *
    FROM exam
    WHERE id=?`,
    [eid]
  );
  await conn.end();

  const exams = (rows as Array<ExamRow>)[0];

  return exams;
}

export async function getStudentExamScore(
  uid: number,
  eid: number
): Promise<number | null> {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT score
    FROM exam_result
    WHERE studentID=? AND examID=?;`,
    [uid, eid]
  );
  await conn.end();

  const score = (rows as Array<ExamResultRow>)[0].score;
  console.log("score", score);

  return score;
}

export async function getTeacherExamScore(eid: number): Promise<ExamResult[]> {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `
    SELECT exam_result.id, studentID, student.name AS studentName, score
    FROM exam_result
    JOIN student ON student.id = exam_result.studentID
    WHERE exam_result.examID = ?;`,
    [eid]
  );
  await conn.end();

  const score = rows as Array<DetailedExamResultRow>;

  return score;
}

export async function updateExamScore(
  erid: number,
  score: number
): Promise<number> {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    `UPDATE exam_result SET score=? WHERE id=?;`,
    [score, erid]
  );
  await conn.end();

  return (rows as ResultSetHeader).affectedRows;
}

export async function insertExamAssign(courseID: number, title: string, location: string, time: string, endtime: string) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'INSERT INTO exam (name, location, time, endtime, courseID) ' +
    'VALUES (?, ?, ?, ?, ?) ',
    [title, location, time, endtime, courseID]
  );
  await conn.end();

  return (rows as ResultSetHeader).insertId
}

export async function insertExamTarget(studentID: number, examID: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'INSERT INTO exam_result (studentID, examID) ' +
    'VALUES (?, ?) ',
    [studentID, examID]
  );
  await conn.end();

  return (rows as ResultSetHeader).affectedRows
}
