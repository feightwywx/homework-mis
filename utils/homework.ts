import { ResultSetHeader } from "mysql2";
import { getNowMySqlDateTime, createMisConnection } from "./mysql";
import { StudentHomework, Homework, HomeworkDetailContent, HomeworkStudentDetail, HomeworkTeacherDetailContent } from "./types";

type StudentHomeworkRow = {
  id: number;
  title: string;
  assignment: string;
  time: string;
  deadline: string;
  completed: number;
  name: string;
}

type HomeworkRow = {
  id: number;
  title: string;
  assignment: string;
  time: string;
  deadline: string;
  name: string;
}

type HomeworkDetailRow = {
  title: string;
  assignment: string;
  time: string;
  deadline: string;
}

type HomeworkDetailContentRow = {
  completed: number;
  time: string;
  content: string;
  score: number;
  comment: string;
}

type TeacherHomeworkDetailRow = {
  cid: number;          // ContentId
  name: string;         // 学生名
  time: string;
  completed: number;
  content: string;
  score: number;
  comment: string;
}

export async function getStudentHomeworks(id: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'SELECT homework.id, title, assignment, homework.time, deadline, completed, teacher.name ' +
    'FROM homework JOIN homework_content JOIN teacher ' +
    'ON homework.id=homework_content.id AND homework.teacherID=teacher.id ' +
    'WHERE studentID=?',
    [id]
  )
  await conn.end();
  
  const homeworks = (rows as Array<StudentHomeworkRow>).map((row) => {
    return {
      ...row,
      completed: row.completed === 0 ? false : true,
      teacher: row.name
    } as StudentHomework
  })

  return homeworks
}

export async function getTeacherHomeworks(id: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'SELECT homework.id, title, assignment, homework.time, deadline, teacher.name ' +
    'FROM homework JOIN teacher ' +
    'ON homework.teacherID=teacher.id ' +
    'WHERE teacherID=?',
    [id]
  )
  await conn.end();
  
  const homeworks = (rows as Array<HomeworkRow>).map((row) => {
    return {
      ...row,
      teacher: row.name
    } as Homework
  })

  return homeworks
}

export async function getHomeworkDetail(id: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'SELECT title, assignment, time, deadline ' +
    'FROM homework ' +
    'WHERE id=?',
    [id]
  )
  await conn.end();
  

  return (rows as Array<HomeworkDetailRow>).length !== 0
    ? (rows as Array<HomeworkDetailRow>)[0]
    : null;
}

export async function getStudentHomeworkDetail(hwid: number, stuid: number) {
  const hwDetail = await getHomeworkDetail(hwid)
  if (hwDetail === null) return null;

  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'SELECT content, time, completed, score, comment ' +
    'FROM homework_content ' +
    'WHERE studentID=? AND homeworkID=?',
    [stuid, hwid]
  );
  await conn.end();
  
  if ((rows as Array<HomeworkDetailContentRow>).length === 0) return null;
  const detailContentRow = (rows as Array<HomeworkDetailContentRow>)[0];

  let accomplishment = undefined;
  if (detailContentRow.time && detailContentRow.content) {
    accomplishment = {
      time: detailContentRow.time,
      content: detailContentRow.content
    };
  }

  let judge = undefined;
  if (detailContentRow.score) {
    judge = {
      score: detailContentRow.score,
      comment: detailContentRow.comment
    }
  }

  const result = {
    detail: hwDetail,
    content: {
      completed: detailContentRow.completed === 0 ? false : true,
      accomplishment,
      judge
    } as HomeworkDetailContent
  } as HomeworkStudentDetail;

  return result;
}

export async function updateHomework(hwid: number, stuid: number, content: string) {
  const now = await getNowMySqlDateTime();

  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'UPDATE homework_content ' +
    'SET content=?, time=?, completed=1 ' +
    'WHERE studentID=? AND homeworkID=?',
    [content, now, stuid, hwid]
  );
  await conn.end();

  return (rows as ResultSetHeader).affectedRows
}

export async function getTeacherHomeworkDetail(hwid: number, tid: number) {
  const hwDetail = await getHomeworkDetail(hwid)
  if (hwDetail === null) return null;

  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'SELECT homework_content.id AS cid, student.`name`, homework_content.time, completed, content, score, comment ' +
    'FROM homework_content LEFT JOIN homework ' +
    'ON homework_content.homeworkID=homework.id ' +
    'LEFT JOIN student ON studentID=student.id ' +
    'WHERE teacherID=? AND homeworkID=?',
    [tid, hwid]
  );
  await conn.end();

  if ((rows as Array<TeacherHomeworkDetailRow>).length === 0) return null;
  const detailContentRow = (rows as Array<TeacherHomeworkDetailRow>);

  return {
    detail: hwDetail,
    content: detailContentRow.map((item) => {
      let accomplishment = undefined;
      if (item.time && item.content) {
        accomplishment = {
          time: item.time,
          content: item.content
        };
      }

      let judge = undefined;
      if (item.score || item.comment) {
        judge = {
          score: item.score,
          comment: item.comment
        }
      }

      return {
        contentId: item.cid,
        studentName: item.name,
        completed: item.completed === 0 ? false : true,
        accomplishment,
        judge
      } as HomeworkTeacherDetailContent;
    })
  }
}

export async function setReject(hwid: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'UPDATE homework_content ' +
    'SET completed=0, score=NULL ' +
    'WHERE homeworkID=?',
    [hwid]
  );
  await conn.end();

  return (rows as ResultSetHeader).affectedRows
}

export async function updateJudge(cid: number, score: number, comment: string) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'UPDATE homework_content ' +
    'SET score=?, comment=? ' +
    'WHERE id=?',
    [score, comment, cid]
  );
  await conn.end();

  return (rows as ResultSetHeader).affectedRows
}

export async function insertAssign(id: number, title: string, assignment: string, deadline: string) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'INSERT INTO homework (title, assignment, deadline, teacherID) ' +
    'VALUES (?, ?, ?, ?) ',
    [title, assignment, deadline, id]
  );
  await conn.end();

  return (rows as ResultSetHeader).insertId
}

export async function insertTarget(hwid: number, target: number) {
  const conn = await createMisConnection();
  const [rows] = await conn.query(
    'INSERT INTO homework_content (completed, studentID, homeworkID) ' +
    'VALUES (0, ?, ?) ',
    [target, hwid]
  );
  await conn.end();

  return (rows as ResultSetHeader).affectedRows
}
