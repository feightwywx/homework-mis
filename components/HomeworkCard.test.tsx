import './matchMedia.mock'
import { render } from '@testing-library/react'
import { Homework } from '../utils/types'
import { HomeworkCard } from './HomeworkCard'

describe('HomeworkCard', () => {
  const homework = {
      id: 1,
      title: 'Homework Title',
      assignment: 'Homework Assignment',
      time: '2022-09-27T02:50:02.000Z',
      deadline: '2022-09-27T02:50:02.000Z',
      teacher: 'Teacher Name',
    } as Homework

    const outdatedHomework = {
      ...homework,
      deadline: '2022-09-27T02:50:02.000Z'
    }

    const completedHomework = {
      ...homework,
      completed: true
    }

    const completedOutdatedHomework = {
      ...homework,
      deadline: '2022-09-27T02:50:02.000Z',
      completed: true
    }

  it('renders HomeworkCard', () => {
    const { container } = render(<HomeworkCard homework={homework} />);
    expect(container).toMatchSnapshot();
  })

  it('renders outdated HomeworkCard', () => {
    const { container } = render(<HomeworkCard homework={outdatedHomework} />);
    expect(container).toMatchSnapshot();
  })

  it('renders completed HomeworkCard', () => {
    const { container } = render(<HomeworkCard homework={completedHomework} />);
    expect(container).toMatchSnapshot();
  })

  it('renders completed outdated HomeworkCard', () => {
    const { container } = render(<HomeworkCard homework={completedOutdatedHomework} />);
    expect(container).toMatchSnapshot();
  })
})
