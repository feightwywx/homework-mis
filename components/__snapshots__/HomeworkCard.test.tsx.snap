// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeworkCard renders HomeworkCard 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
    style="margin-bottom: 16px; display: block;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Homework Title
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/homeworkDetail/1"
          >
            查看
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <span
        class="ant-typography"
      >
        发布时间：
        2022/9/27 10:50:02
      </span>
      <br />
      <span
        class="ant-typography"
      >
        截止时间：
        2222/9/27 10:50:02
      </span>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
        style="margin: 16px 0px;"
      />
      <div>
        <span
          class="ant-avatar ant-avatar-sm ant-avatar-circle ant-avatar-icon"
          style="margin-top: -4px;"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ant-typography"
          style="margin-left: 12px;"
        >
          <b>
            Teacher Name
          </b>
          布置的作业
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeworkCard renders completed HomeworkCard 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
    style="margin-bottom: 16px; display: block;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Homework Title
          <span
            class="ant-tag ant-tag-success"
            style="margin-left: 8px; vertical-align: middle;"
          >
            <span
              aria-label="check-circle"
              class="anticon anticon-check-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="check-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </span>
            <span>
              已完成
            </span>
          </span>
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/homeworkDetail/1"
          >
            查看
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <span
        class="ant-typography"
      >
        发布时间：
        2022/9/27 10:50:02
      </span>
      <br />
      <span
        class="ant-typography"
      >
        截止时间：
        2222/9/27 10:50:02
      </span>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
        style="margin: 16px 0px;"
      />
      <div>
        <span
          class="ant-avatar ant-avatar-sm ant-avatar-circle ant-avatar-icon"
          style="margin-top: -4px;"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ant-typography"
          style="margin-left: 12px;"
        >
          <b>
            Teacher Name
          </b>
          布置的作业
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeworkCard renders completed outdated HomeworkCard 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
    style="margin-bottom: 16px; display: block;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Homework Title
          <span
            class="ant-tag ant-tag-success"
            style="margin-left: 8px; vertical-align: middle;"
          >
            <span
              aria-label="check-circle"
              class="anticon anticon-check-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="check-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </span>
            <span>
              已完成
            </span>
          </span>
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/homeworkDetail/1"
          >
            查看
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <span
        class="ant-typography"
      >
        发布时间：
        2022/9/27 10:50:02
      </span>
      <br />
      <span
        class="ant-typography"
      >
        截止时间：
        2022/9/27 10:50:02
      </span>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
        style="margin: 16px 0px;"
      />
      <div>
        <span
          class="ant-avatar ant-avatar-sm ant-avatar-circle ant-avatar-icon"
          style="margin-top: -4px;"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ant-typography"
          style="margin-left: 12px;"
        >
          <b>
            Teacher Name
          </b>
          布置的作业
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeworkCard renders outdated HomeworkCard 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
    style="margin-bottom: 16px; display: block;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Homework Title
          <span
            class="ant-tag"
            style="margin-left: 8px; vertical-align: middle;"
          >
            已过期
          </span>
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/homeworkDetail/1"
          >
            查看
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <span
        class="ant-typography"
      >
        发布时间：
        2022/9/27 10:50:02
      </span>
      <br />
      <span
        class="ant-typography"
      >
        截止时间：
        2022/9/27 10:50:02
      </span>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
        style="margin: 16px 0px;"
      />
      <div>
        <span
          class="ant-avatar ant-avatar-sm ant-avatar-circle ant-avatar-icon"
          style="margin-top: -4px;"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ant-typography"
          style="margin-left: 12px;"
        >
          <b>
            Teacher Name
          </b>
          布置的作业
        </span>
      </div>
    </div>
  </div>
</div>
`;
